src/runtime/test_case_runner.js

```diff
@@ -166,26 +172,40 @@ export default class TestCaseRunner {
 
   async run() {
     this.emitPrepared()
-    this.emit('test-case-started', {})
-    await this.runHooks(
-      this.beforeHookDefinitions,
-      {
-        sourceLocation: this.testCaseSourceLocation,
-        pickle: this.testCase.pickle,
-      },
-      true
-    )
-    await this.runSteps()
-    await this.runHooks(
-      this.afterHookDefinitions,
-      {
-        sourceLocation: this.testCaseSourceLocation,
-        pickle: this.testCase.pickle,
-        result: this.result,
-      },
-      false
-    )
-    this.emit('test-case-finished', { result: this.result })
+    let retries = 0
+    for (; retries <= this.retries; retries++) {
+      this.emit('test-case-started', {})
+      if (retries > 0) {
+        this.resetTestProgressData()
+        this.result.retryAttempt = retries
```

Comment:
I'd rather have this be a top level property on test-case-started and test-case-finished instead of in the result. Thoughts on naming it attemptNumber as well and always passing it?
```js
// Set this.maxAttempts = 1 + retries in the constructor
let attemptNumber = 1;
this.emit('test-case-started', {attemptNumber})
this.emit('test-case-finished', {attemptNumber, result: this.result})
```

Reply:
Done